docs/code-examples/build.rs

```diff
@@ -0,0 +1,101 @@
+//! Finds all the `*.rs` files in `docs/code-examples/all`,
+//! copies them to `src` (with slight modifications), and generate a `lib.rs` for them.
+//!
+//! The reason we combine all the examples into a single binary
+//! is to reduce the amount of binaries in our workspace.
+//!
+//! Motivation: <https://github.com/rerun-io/rerun/issues/4623>
+
+use std::{fs, path::Path};
+
+use itertools::Itertools as _;
+use rust_format::Formatter as _;
+
+fn main() {
+    let path = Path::new("./all");
+
+    assert!(path.exists() && path.is_dir(), "Failed to find {path:?}");
+
+    let mut examples = Vec::new();
+
+    for entry in fs::read_dir(path).unwrap().flatten() {
+        let path = entry.path();
+        if let Some(extension) = path.extension() {
+            if extension == "rs" {
+                let example_name = path.file_stem().unwrap().to_str().unwrap().to_owned();
+
+                let contents = fs::read_to_string(&path).unwrap();
+
+                // some examples lack a main, see https://github.com/rerun-io/landing/issues/515
```

Comment:
```suggestion
                // TODO(#515): some examples lack a main, they should come with their necessary stub code commented out so that we can re-add it here.
```